src/code42cli/cmds/securitydata.py

```diff
@@ -362,84 +337,92 @@ def search(
+                "fileOwner",
+                "md5Checksum",
+                "sha256Checksum",
+                "riskIndicators",
+                "riskSeverity",
+            ]
+
     if use_checkpoint:
-        checkpoint_name = use_checkpoint
-        # if checkpoint name exists, checkpoint should be that eventId,
-        # otherwise it should set the initial value to ""
-        checkpoint = cursor.get(checkpoint_name) or ""
-
-        # older app versions stored checkpoint as float timestamp.
-        # we handle those here until the next run containing events will store checkpoint as the last eventId
-        try:
-            state.search_filters.append(
-                InsertionTimestamp.on_or_after(float(checkpoint))
-            )
-            checkpoint = ""
-        except ValueError:
-            pass
+        cursor = _get_file_event_cursor_store(state.profile.name)
+        checkpoint = _handle_timestamp_checkpoint(cursor.get(use_checkpoint), state)
+
+        def checkpoint_func(event):
+            cursor.replace(use_checkpoint, event["eventId"])
+
     else:
-        checkpoint = ""
+        checkpoint = checkpoint_func = None
 
     query = _construct_query(state, begin, end, saved_search, advanced_query, or_query)
-    events = _get_all_file_events(state, query, checkpoint)
+    dfs = _get_all_file_events(state, query, checkpoint)
+    formatter = FileEventsOutputFormatter(format, checkpoint_func=checkpoint_func)
+    # sending to pager when checkpointing can be inaccurate due to pager buffering, so disallow pager
+    force_no_pager = use_checkpoint
+    formatter.echo_formatted_dataframes(
+        dfs, columns=columns, force_no_pager=force_no_pager
+    )
 
-    if use_checkpoint:
-        checkpoint_name = use_checkpoint
-        events = _store_updated_checkpoint(cursor, checkpoint_name, events)
 
-    events_list = []
-    for event in events:
-        events_list.append(event)
-    if not events_list:
-        click.echo("No results found.")
-        return
-    formatter.echo_formatted_list(events_list)
+@security_data.command(cls=SendToCommand)
+@file_event_options
+@search_options
+@sdk_options()
+@searchopt.server_options
+@searchopt.send_to_format_options
+def send_to(
+    state,
+    begin,
+    end,
+    advanced_query,
+    use_checkpoint,
+    saved_search,
+    or_query,
+    columns,
+    **kwargs,
+):
+    """Send events to the given server address.
 
+    HOSTNAME format: address:port where port is optional and defaults to 514.
+    """
+    if use_checkpoint:
+        cursor = _get_file_event_cursor_store(state.profile.name)
+        checkpoint = _handle_timestamp_checkpoint(cursor.get(use_checkpoint), state)
 
-def _construct_query(state, begin, end, saved_search, advanced_query, or_query):
+        def checkpoint_func(event):
+            cursor.replace(use_checkpoint, event["eventId"])
 
-    if advanced_query:
-        state.search_filters = advanced_query
-    if saved_search:
```

Comment:
I don't think this really matters - ideally only exactly one of these would be true at a time, but this could probably be `elif` ? I think `advanced-search`, `saved-search` and `begin/end` are all exclusive from each other